src/main/java/io/airlift/slice/Slice.java

```diff
@@ -97,9 +99,21 @@ public static Slice toUnsafeSlice(ByteBuffer byteBuffer)
     private final long retainedSize;
 
     /**
-     * Reference is typically a ByteBuffer object, but can be any object this
-     * slice must hold onto to assure that the underlying memory is not
-     * freed by the garbage collector.
+     * Reference have two use cases:
+     *
+     * 1. It can be an object this slice must hold onto to assure that the
+     * underlying memory is not freed by the garbage collector.
+     * It is typically a ByteBuffer object, but can be any object.
+     *
+     * 2. If reference is not used to prevent garbage collector to free
+     * underlying memory, it will be used as the flag to indicate whether
```

Comment:
Grammar: "to indicate if the slice is compact"

(I'm not sure the original is wrong, but it sounds a bit funny. "whether or not" would read better, but it can be shortened to "if")